lib/render/game.php

```diff
@@ -259,3 +259,43 @@ function RenderRecentGamePlayers($recentPlayerData): void
     echo "</tbody></table>";
     echo "</div>";
 }
+
+function RenderGameProgress(int $numAchievements, int $numEarnedCasual, int $numEarnedHardcore)
+{
+    $pctComplete = 0;
+    $pctHardcore = 0;
+    $pctHardcoreProportion = 0;
+    $title = '';
+
+    if ($numAchievements) {
+        $pctAwardedCasual = ($numEarnedCasual + $numEarnedHardcore) / $numAchievements;
+        $pctAwardedHardcore = $numEarnedHardcore / $numAchievements;
+        $pctAwardedHardcoreProportion = 0;
+        if ($numEarnedHardcore > 0) {
+            $pctAwardedHardcoreProportion = $numEarnedHardcore / ($numEarnedHardcore + $numEarnedCasual);
+        }
+
+        $pctComplete = sprintf("%01.0f", $pctAwardedCasual * 100.0);
```

Comment:
I think the fractions can be omitted as it's dynamically typed. Not that important though.